nemo/utils/exp_manager.py

```diff
@@ -83,19 +109,101 @@ def exp_manager(trainer: 'pytorch_lightning.Trainer', cfg: Optional[Union[DictCo
     if cfg is None:
         logging.error("exp_manager did not receive a cfg argument. It will be disabled.")
         return
+
+    # Ensure passed cfg is compliant with ExpManagerConfig
     schema = OmegaConf.structured(ExpManagerConfig)
     if isinstance(cfg, dict):
         cfg = OmegaConf.create(cfg)
     elif not isinstance(cfg, DictConfig):
         raise ValueError(f"cfg was type: {type(cfg)}. Expected either a dict or a DictConfig")
-    # Ensure passed cfg is compliant with ExpManagerConfig
     cfg = OmegaConf.merge(schema, cfg)
 
-    if get_original_cwd() != os.getcwd():
+    error_checks(trainer, cfg)  # Ensures that trainer options are compliant with NeMo and exp_manager arguments
+
+    log_dir, root_dir, name, version = get_log_dir(
+        trainer=trainer,
+        root_dir=cfg.root_dir,
+        name=cfg.name,
+        version=cfg.version,
+        explicit_log_dir=cfg.explicit_log_dir,
+        use_datetime_version=cfg.use_datetime_version,
+        resume=cfg.resume,
+        previous_log_dir=cfg.previous_log_dir,
+    )
+    cfg.name = name
```

Comment:
Merging `cfg = OmegaConf.merge(schema, cfg)` results in cfg.name being None if it does not exist in the cfg passed to exp_manager.

If it did not exist in the dictionary, I'm sure some of the tests would fail with an OmegaConfException rather than succeed..

Reply:
Interesting, so the merge can add args but crashes iff args mismatch. Good to know.